src/app/services/baw-api/bookmark/bookmarks.service.ts

```diff
@@ -32,9 +34,18 @@ export class BookmarksService extends StandardApi<Bookmark> {
   list(): Observable<Bookmark[]> {
     return this.apiList(endpoint(Empty, Empty));
   }
-  filter(filters: Filters): Observable<Bookmark[]> {
+  filter(filters: Filters<Bookmark>): Observable<Bookmark[]> {
     return this.apiFilter(endpoint(Empty, Filter), filters);
   }
+  filterByCreator(
+    filters: Filters<Bookmark>,
+    user?: IdOr<User>
+  ): Observable<Bookmark[]> {
+    return this.apiFilter(
+      endpoint(Empty, Filter),
+      user ? filterById<Bookmark>(filters, "creatorId", user) : filters
```

Comment:
reading this from GH I can't confirm, but that `creatorId` key is strongly typed? I think it is, nice!

Reply:
Yup, its strongly typed

Reply:
Only thing it doesn't do is validate that the key provided is of type Id, specifically because I'm unsure how generic I want to make the function.